src/game/game_init.c

```diff
@@ -690,6 +702,15 @@ void init_controllers(void) {
             gControllers[cont++].controllerData = &gControllerPads[port];
         }
     }
+    if (__osControllerTypes[1] == CONT_TYPE_GCN) {
+        gGamecubeControllerPort = 1;
+        gPlayer1Controller = &gControllers[1];
+    } else {
+        if (__osControllerTypes[0] == CONT_TYPE_GCN) {
+            gGamecubeControllerPort = 0;
+        }
+        gPlayer1Controller = &gControllers[0];
+    }
```

Comment:
I don't get the point of this code, it seems like a personal edge case that might cause issues for other people or potentially emulators. It also wouldn't work with more than one GC controller connected, and it might potentially hinder the ability to use a dual controller configuration with something like analog camera.

Reply:
Looks like this unfortunately might be necessary actually, but it should still get its own console check then.